project2/py/web_scraper.py

```diff
@@ -45,6 +46,42 @@ def parse_ingredients(ingredients):
     objs = utils.pmap(Ingredient, ingredients)
     return objs
 
+def parse_methods(directions, kb):
```

Comment:
`parse_methods` and `parse_tools` look eerily similar. See `extract_tools` for a similar suggestion.